app/assets/scripts/components/project-card.js

```diff
@@ -14,22 +14,33 @@ function categoryLink (base, categoryName) {
 }
 
 function isOntime (project) {
-  let planned = parseProjectDate(project.planned_end_date);
-  let actual = parseProjectDate(project.actual_end_date);
+  let plannedEnd = parseProjectDate(project.planned_end_date);
+  let actualEnd = parseProjectDate(project.actual_end_date);
+  let plannedStart = parseProjectDate(project.planned_start_date);
+  let actualStart = parseProjectDate(project.actual_start_date);
 
-  // if there's no actual date, and the planned date is still in the future,
-  // check if there are start dates to use instead.
-  if (!actual && planned && planned > new Date().getTime()) {
-    planned = parseProjectDate(project.planned_start_date);
-    actual = parseProjectDate(project.actual_start_date);
-  }
-
-  if (!actual || !planned) {
+  if (!actualStart || !plannedStart) {
     return null;
-  } else if (actual > planned) {
-    return false;
+    // project incomplete and late
+  } else if ((actualStart <= plannedStart) && (!actualEnd && plannedEnd && plannedEnd < new Date().getTime())) {
+    return 'extended';
+    // project completed late
+  } else if ((actualStart <= plannedStart) && (actualEnd > plannedEnd)) {
+    return 'extended';
+    // project in progress, started late
+  } else if (((actualStart > plannedStart) && (!actualEnd && plannedEnd && plannedEnd > new Date().getTime()))) {
+    return 'delayed';
+    // project completed, started late
+  } else if (((actualStart > plannedStart) && (actualEnd <= plannedEnd))) {
+    return 'delayed';
+    // project both delayed and completed late
+  } else if (((actualStart > plannedStart) && (actualEnd > plannedEnd))) {
+    return 'delayed';
+    // project in progress, both delayed and late
+  } else if (((actualStart > plannedStart) && (!actualEnd && plannedEnd && plannedEnd < new Date().getTime()))) {
+    return 'delayed';
```

Comment:
I wasn't happy with the duplication of return values either, but my intention was readability. I also could have made any of the overlapping conditionals into ORs, but I didn't think that doing so would result in any real performance gains and it would have made it harder to read/ maintain. Storing each duplicated comparison is a good idea, though; I'll see what I can do there.